class-components/src/components/DetailCardInfo/DetailCardInfo.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useState } from 'react';
+
+import { BASE_URL } from '../../constants';
+import { ISingleResult, IDetailCardInfoProps } from '../../types/AppTypes';
+import './DetailCardInfo.css';
+
+const DetailCardInfo: React.FC<IDetailCardInfoProps> = ({ itemId, onClose }) => {
+  const [detailInfo, setDetailInfo] = useState<ISingleResult | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchDetailInfo = async () => {
+      setIsLoading(true);
+      try {
+        const resp = await fetch(`${BASE_URL}people/${itemId}/`);
+        if (!resp.ok) {
+          throw new Error('Failed to fetch detail info');
+        }
+        const data: ISingleResult = await resp.json();
+        setDetailInfo(data);
+        setIsLoading(false);
+      } catch (error) {
+        console.error('Error fetching detail info:', error);
+        setIsLoading(false);
+      }
+    };
+
+    fetchDetailInfo();
+  }, [itemId]);
+
+  if (isLoading) {
+    return <div className="loading loading_detail-info">Loading detail info...</div>;
+  }
+
+  if (!detailInfo) {
+    return <div className="detail-card-info">Detail info not found.</div>;
+  }
+
+  return (
+    <div className="detail-card-info">
+      <div className="detail-card-info-close">
+        <button type="button" onClick={onClose} aria-label="Close">
+          X
+        </button>
+      </div>
+      <div className="detail-header">
+        <img
+          className="detail-info-img"
+          src={`https://starwars-visualguide.com/assets/img/characters/${detailInfo.url.split('/')[5]}.jpg`}

```

Comment:
same as in Card component, use useMemo with detailInfo dependency instead of this `detailInfo.url.split('/')[5]`